addons/website/models/website_visitor.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+from odoo.http import request
+from datetime import datetime, timedelta
+from hashlib import md5
+
+
+class WebsiteVisitor(models.Model):
+    _name = 'website.visitor'
+    _description = 'Website Visitor'
+    _order = 'last_connection_date DESC'
+
+    name = fields.Char('Name', default='Website Visitor')
+    website_id = fields.Many2one('website', "Website")
+    partner_ids = fields.Many2many('res.partner')
+    partner_count = fields.Integer('# Partners', compute="_compute_partner_count")
+    create_date = fields.Datetime('First connexion date', default=fields.Datetime.now)
+    last_connection_date = fields.Datetime('Last Connection', help="Last page view date")
+    country_id = fields.Many2one('res.country', 'Country', readonly=True)
+    country_flag = fields.Binary(related="country_id.image", string="Country Flag")
+    lang_id = fields.Many2one('res.lang', string='Language', help="Language from the website when visitor has been created")
+    visit_count = fields.Integer('Number of visits', default=1, help="If last visit was more than 8 hours ago, the new connection is considered as a new visit. Else, it is considered as same visit.")
+    page_ids = fields.One2many('website.visitor.page', 'visitor_id', string='Visited Pages History')
+    page_view_count = fields.Integer('Page Views', compute="_compute_page_view_count")
+    unique_page_ids = fields.Many2many('website.page', string="Visited Pages", compute="_compute_unique_page_ids", store=True)
+    time_last_action = fields.Float('Time since last action', compute="_compute_time_last_action", help='Time since last page view in seconds')
+    is_connected = fields.Boolean('Is connected ?', compute='_compute_is_connected', help='A visitor is considered as connected if his last page view was within the last 5 minutes.')
+    # Maybe not necessary, only used for visitor - lead - customer views (filtered on type)
+    # but could check directly on domain.
+    type = fields.Selection([
```

Comment:
```suggestion
    visitor_type = fields.Selection([
```
please, see guidelines https://github.com/odoo/odoo/commit/cfaf1f5ac5b634f32b6a3b65f54bf2e6761ec77c